Simplify deprecation policy (#4259)
FusionAuth/fusionauth-site: astro/src/content/docs/operate/roadmap/deprecation-policy.mdx

description: Learn more about the FusionAuth deprecation policy.
---

> FusionAuth is an architectural component integrated into your applications or systems. Such integration means that when changes to FusionAuth occur, it can negatively affect your application. As developers ourselves, we strive to maintain backwards compatibility. We will not remove API support in minor or patch versions (only in major versions), except for security/performance reasons noted below.

Reviewer: Considering we have only ever released one major version, should this instead say:

```suggestion
FusionAuth is an architectural component integrated into your applications or systems. Such integration means that when changes to FusionAuth occur, it can negatively affect your application. As developers ourselves, we strive to maintain backwards compatibility. Only minor and major versions will change API support (never patch releases).
```

suggestion: I always find it easier to understand conditionals like this when we avoid negatives
suggestion: if we extend this to 'major and minor', maybe we can omit the consideration about security and performance reasons?
nitpick: avoid slashes, preferring 'and' and 'or' when possible

Author: @nathan-contino check out revision

Before removing a feature, the FusionAuth team takes the following steps:

* In the first major or point release where the behavior is deprecated, the deprecation is noted in the release notes. The feature is also marked as deprecated in the documentation.

> * In a major release after the behavior is deprecated, the feature and corresponding documentation are removed. When the feature is removed, it is noted in the release notes. The documentation for the feature is also removed.

Reviewer: Does this mean 'one major release after' or is this a noncommittal 'some major release after'? It might be better to be explicit here.

Author: @nathan-contino check out revision

> * In a major release after the behavior is deprecated, the feature and corresponding documentation are removed. When the feature is removed, it is noted in the release notes. The documentation for the feature is also removed.

Reviewer 2: This is confusing to me because we don't do major releases. We only do minor and patch releases. 
```suggestion
* In a minor release after the behavior is deprecated, the feature and corresponding documentation are removed. When the feature is removed, it is noted in the release notes. The documentation for the feature is also removed.
```

Reviewer 2: I'd also prefer to keep the 3 month minimum timeframe, but I suppose that is an engineering decision.
